pytorch_lightning/trainer/training_loop.py

```diff
@@ -735,8 +752,13 @@ def train_step_and_backward_closure():
 
     @contextmanager
     def block_ddp_sync_behaviour(self):
-        if isinstance(self.trainer.model, torch.nn.parallel.DistributedDataParallel):
-            yield self.trainer.model.no_sync()
+        """
+        Blocks ddp sync gradients behaviour on backwards pass.
+        This is useful for skipping sync when accumulating gradients, reducing communication overhead
+        Returns: context manager with sync behaviour off
+        """
+        if self.trainer.accelerator_backend is not None:
+            yield self.trainer.accelerator_backend.block_ddp_plugin_sync_behaviour()
         else:
             yield
```

Comment:
```suggestion
        cm = None
        if self.trainer.accelerator_backend is not None:
            cm = self.trainer.accelerator_backend.block_ddp_plugin_sync_behaviour()
        yield cm
```

Reply:
It is easier to read in the previous form.